apps/native-component-list/src/screens/SensorScreen.tsx

```diff
@@ -201,6 +202,48 @@ class LightSensor extends SensorBlock<Sensors.LightSensorMeasurement> {
   );
 }
 
+const PedometerSensor = () => {
+  const [isPedometerAvailable, setIsPedometerAvailable] = useState('checking');
+  const [pastStepCount, setPastStepCount] = useState(0);
+  const [currentStepCount, setCurrentStepCount] = useState(0);
+
+  useEffect(() => {
+    const subscribe = async () => {
+      const isAvailable = await Sensors.Pedometer.isAvailableAsync();
+      setIsPedometerAvailable(String(isAvailable));
+
+      if (isAvailable) {
+        const end = new Date();
+        const start = new Date();
+        start.setDate(end.getDate() - 1);
+
+        const pastStepCountResult = await Sensors.Pedometer.getStepCountAsync(start, end);
+        if (pastStepCountResult) {
+          setPastStepCount(pastStepCountResult.steps);
+        }
+
+        return Sensors.Pedometer.watchStepCount((result) => {
+          setCurrentStepCount(result.steps);
+        });
+      }
+      
+      return null;
+    };
+
+    const listener = subscribe();
+    return () => listener?.remove();
```

Comment:
```suggestion
    const subscription = subscribe();
    return () => subscription.then(listener => listener?.remove());
```
Sorry, I didn't notice that `subscribe` is async. Maybe something like this would work?

Reply:
It wouldn't because we can't return a function that returns a promise inside an useEffect